Classes/Utility/ImageUtility.php

```diff
@@ -259,13 +259,17 @@ protected static function getBasicMetadata(string $fileName): array
      */
     protected static function safeUtf8Encode(?string $text): ?string
     {
+        if (function_exists('mb_convert_encoding') === false) {
+            return $text;
+        }
+
         if (function_exists('mb_detect_encoding')) {
             if (mb_detect_encoding($text, 'UTF-8', true) !== 'UTF-8') {
-                $text = utf8_encode($text);
+                $text = mb_convert_encoding($text, 'UTF-8', 'ISO-8859-1');
```

Comment:
I (partly) agree with that very part, the rest is imho incorrect tu properly support various configurations.

IMHO best would simply to copy over what I wrote in another extension of mine for the very same conversion function:

https://github.com/xperseguers/t3ext-extractor/blob/master/Classes/Service/Php/PhpService.php#L625-L640